### ISSUE TITLE
False-positive `call-arg`: Too many arguments for "enum.auto"

### ISSUE THREAD

**Issue by bersbersbers** [python/mypy#16980]
**Bug Report**

This code works (A is 5, B is 6), but mypy reports an error:

**To Reproduce**

```python
from enum import IntEnum, auto

class MyEnum(IntEnum):
    A = auto(5)
    B = auto()

print(MyEnum.A)
print(MyEnum.B)
```
https://mypy-play.net/?mypy=latest&python=3.12&gist=09173111e5af20a50c0a8ab24223ef61

**Expected Behavior**

No error.

**Actual Behavior**

> bug.py:5: error: Too many arguments for "auto"  [call-arg]
Found 1 error in 1 file (checked 1 source file)

**Your Environment**

- Mypy version used: mypy 1.8.0 (compiled: yes)
- Mypy command-line flags: none
- Python version used: 3.12.2
